src/AppendStream.php

```diff
@@ -92,8 +95,6 @@ public function close()
      * Detaches each attached stream.
      *
      * Returns null as it's not clear which underlying stream resource to return.
-     *
-     * {@inheritdoc}
      */
     public function detach()
```

Comment:
I see that you are adding return types.. why not in all methods like this one?

Reply:
I didn't touch the psr-7 methods. Want to do that in a next step.

Reply:
added return types to psr-7 implementations in a new commit.
this is safe due to phps contravariant return type rules.